serve_test.go

```diff
@@ -637,7 +637,7 @@ func TestFlow(t *testing.T) {
 		assert.Equal(t, crlf(`{"ans":"Hello World"}`), w.Body.String())
 	})
 	t.Run("with nil return in ErrorHandler", func(t *testing.T) {
-		e := NewEngine(WithErrorHandler(func(err error) error { return nil }))
+		e := NewEngine(WithErrorHandler(func(_ context.Context, err error) error { return nil }))
 		w := httptest.NewRecorder()
 		ctx := newTestCtx(w, httptest.NewRequest("GET", "/", nil))
```

Comment:
I would expect to see some tests to check the context received by the ErrorHandler is the one from the request

Either here or in engine_test.go, or both

Reply:
Could you try explaining 

> I would expect to see some tests to check the context received by the ErrorHandler is the one from the request

In a different way? 

Like ensuring it's from `http.Request`??

Reply:
🤔 I guess. I could call `HTTPHandler` and slogassert the whole chain

Reply:
I was talking about something like this

```go
		var receivedCtx context.Context
		e := NewEngine(WithErrorHandler(func(ctx context.Context, err error) error {
			receivedCtx = ctx
			return nil
		}))
 		w := httptest.NewRecorder()
 		ctx := newTestCtx(w, httptest.NewRequest("GET", "/", nil))
 		Flow(e, ctx, testControllerWithError)
		assert.Equal(t, ctx, receivedCtx)
```

Reply:
Ah, cool. Nice idea. 

I think I'll do both as it's pretty easy to add a log in Flow and not have it be `***Context`